pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -101,167 +103,132 @@ func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfi
+			pluginList: reflect.ValueOf(&f.postBindPlugins),
+			pluginSet:  plugins.PostBind,
+			pluginType: reflect.TypeOf((*PostBindPlugin)(nil)).Elem(),
+		},
+		{
+			name:       "Unreserve",
+			pluginList: reflect.ValueOf(&f.unreservePlugins),
+			pluginSet:  plugins.Unreserve,
+			pluginType: reflect.TypeOf((*UnreservePlugin)(nil)).Elem(),
+		},
+		{
+			name:       "Permit",
+			pluginList: reflect.ValueOf(&f.permitPlugins),
+			pluginSet:  plugins.Permit,
+			pluginType: reflect.TypeOf((*PermitPlugin)(nil)).Elem(),
+		},
+		{
+			name:       "QueueSort",
+			pluginList: reflect.ValueOf(&f.queueSortPlugins),
+			pluginSet:  plugins.QueueSort,
+			pluginType: reflect.TypeOf((*QueueSortPlugin)(nil)).Elem(),
+		},
+	}
+
+	for _, extensionPoint := range extensionPoints {
+		if err := updatePluginList(
+			extensionPoint.name,
+			extensionPoint.pluginList,
+			extensionPoint.pluginSet,
+			extensionPoint.pluginType,
+			pluginsMap,
+		); err != nil {
+			return nil, err
+		}
+	}
+
+	for _, scorePlugin := range f.scorePlugins {
+		if f.pluginNameToWeightMap[scorePlugin.Name()] == 0 {
+			return nil, fmt.Errorf("score plugin %q is not configured with weight", scorePlugin.Name())
+		}
+	}
+
+	if len(f.queueSortPlugins) > 1 {
+		return nil, fmt.Errorf("only one queue sort plugin can be enabled")
 	}
 
-	if plugins.PostFilter != nil {
-		for _, r := range plugins.PostFilter.Enabled {
-			if pg, ok := pluginsMap[r.Name]; ok {
-				p, ok := pg.(PostFilterPlugin)
-				if !ok {
-					return nil, fmt.Errorf("plugin %q does not extend post-filter plugin", r.Name)
-				}
-				f.postFilterPlugins = append(f.postFilterPlugins, p)
-			} else {
-				return nil, fmt.Errorf("post-filter plugin %q does not exist", r.Name)
-			}
-		}
-	}
+	return f, nil
+}
 
-	if plugins.PreBind != nil {
-		for _, pb := range plugins.PreBind.Enabled {
-			if pg, ok := pluginsMap[pb.Name]; ok {
-				p, ok := pg.(PreBindPlugin)
-				if !ok {
-					return nil, fmt.Errorf("plugin %q does not extend prebind plugin", pb.Name)
-				}
-				f.preBindPlugins = append(f.preBindPlugins, p)
-			} else {
-				return nil, fmt.Errorf("prebind plugin %q does not exist", pb.Name)
-			}
-		}
+func updatePluginList(name string, pluginList reflect.Value, pluginSet *config.PluginSet, pluginType reflect.Type, pluginsMap map[string]Plugin) error {
+	if pluginSet == nil {
+		return nil
 	}
 
-	if plugins.Bind != nil {
-		for _, pb := range plugins.Bind.Enabled {
-			if pg, ok := pluginsMap[pb.Name]; ok {
-				p, ok := pg.(BindPlugin)
-				if !ok {
-					return nil, fmt.Errorf("plugin %q does not extend bind plugin", pb.Name)
-				}
-				f.bindPlugins = append(f.bindPlugins, p)
-			} else {
-				return nil, fmt.Errorf("bind plugin %q does not exist", pb.Name)
-			}
+	plugins := pluginList.Elem()
+	set := sets.NewString()
+	for _, ep := range pluginSet.Enabled {
+		pg, ok := pluginsMap[ep.Name]
+		if !ok {
+			return fmt.Errorf("%s plugin %q does not exist", pluginType.String(), ep.Name)
```

Comment:
Remove the word "plugin" from the error message, it is redundant because pluginType.String() will be something like FilterPlugin and so it has the suffix plugin already.

```"%s %q does not exist"```